### ISSUE TITLE
Unable to find timestamp for a transcribed audio for python

### ISSUE THREAD

**Issue by MHumza3656** [facebookresearch/flashlight#502]
Hello everyone,
I'm trying to find a solution for timestamps for specific audio file. I have implemented the Colab example for transcription which has a bug for transcribing longer audios but I've worked my way with that through batching the audio.

Now, I'm trying to find timestamps for an audio file. Please recommend me a source on how to do it in flashlight? if not possible through that shall I import any script from wav2letter? Any guidance will be highly appreciated, thank you!

**Comment by tlikhomanenko** [comment]
Hey, please check in this tutorial https://github.com/facebookresearch/flashlight/blob/master/flashlight/app/asr/tutorial/notebooks/InferenceAndAlignmentCTC.ipynb alignment section "Check align result" which shows the output file of the alignment with timestamps on each token.

Or do you mean something else?

**Comment by MHumza3656** [comment]
Right, after writing this response I was looking around and found the solution within "Check Align". This is amazing, but the issue with this is that it requires a separate call for getting the timestamps.

Do we have a parameter within the inference method that we can pass and the model not only returns the transcribed text but a JSON file that gives transcription and a word to word time stamps where the start and end time is mentioned for each word. An example is:
`"words": [{"startTime": "0s", "endTime": "6.400s", "word": "Brooke"}, {"startTime": "6.400s", "endTime": "6.500s", "word": "is"}, {"startTime": "6.500s", "endTime": "7s", "word": "daunting"}`

**Comment by tlikhomanenko** [comment]
You can merge two cpp files to do so, we didn't do them jointly (you can do whatever format you want then).

**Comment by MHumza3656** [comment]
oh interesting, since the InferenceAndAlignmentCTC notebook that uses these two files "fl_asr_tutorial_inference_ctc" and "fl_asr_align" are two files that are present in some bytecode I believe and I can't read them, Could you point me to the files they use? Secondly, will changing those files will affect the files I use? or will I have to recompile these files?

**Comment by tlikhomanenko** [comment]
The source cpp of these binaries are here https://github.com/facebookresearch/flashlight/tree/master/flashlight/app/asr/tutorial, see in this dir *cpp files. You can change them (even in colab) and then you need to rerun `make` command only (no need to rerun cmake). So you can copy/emulate functionality from one file to another, rerun cmake and use only one binary.

**Comment by MHumza3656** [comment]
awesome thank you so much
I have just one optional question and I believe the answer to it is "Yes" as well. Is code for both "fl_asr_tutorial_inference_ctc" & "fl_asr_align" written in InferenceCTC.cpp?
Again, thank you for all your help. Much appreciated

**Comment by tlikhomanenko** [comment]
Ahh, sorry, forgot about alignment (messed it with finetune). It is here https://github.com/facebookresearch/flashlight/tree/master/flashlight/app/asr/tools/alignment - Align.cpp. So fl_asr_tutorial_inference_ctc and  fl_asr_align are two different cpp files and we compile each cpp to its own binary. Because mainly align is a tool and very general (can be used for VAD or other tasks).

**Comment by MHumza3656** [comment]
Absolutely makes sense, can I custom combine these files and expect it to work or will there be other file dependencies. Mainly my idea is to have another parameter in "fl_asr_tutorial_inference_ctc" that calls "fl_asr_align" so instead of loading the model separately and inferencing again it does all of it in one go.
You also might have a better understanding of should I invest my time in this or go with the current flow? I have made this current functionality work to my usecase

**Comment by tlikhomanenko** [comment]
Everything is in these two files, so just combine them. 

If you managed to set the workflow, then go ahead with him, otherwise you need to understand a bit of code and be sure you can program in c++ =).

**Comment by MHumza3656** [comment]
@tlikhomanenko just for better understanding, could you please tell me what are these files are compiled? `fl_asr_tutorial_inference_ctc` and `fl_asr_align`. I have implemented them in jupyter notebook and they work just fine but not in Pycharm, so trying to better understand them. Thank you

**Comment by tlikhomanenko** [comment]
Sure, here it is definition https://github.com/flashlight/flashlight/blob/master/flashlight/app/asr/CMakeLists.txt#L67-L68.

**Comment by RuABraun** [comment]
Is there no method to get an alignment as part of the python based decoding? It seems wasteful to have to make a separate call when the timing information should already be contained within the decoding graph.
